content/design-patterns/read-receipts.textile

```diff
@@ -0,0 +1,63 @@
+---
+title: Message Read Receipts
+section: design-patterns
+index: 32
+languages:
+  - none
+---
+
+Most of the chat application users expect delivery or read receipts on the messages they sent. WhatsApp was the first to introduce this feature back in 2014, while almost all of the other chat apps in the market soon followed suit. With this Design Pattern, you'll understand how to implement read receipts in your chat apps using "Ably's Realtime Platform":https//www.ably.io.
+
+h2(#what). What are read receipts?
+
+Continuing on WhatsApp's example, we know that when we send a message via WhatsApp, we see a single grey tick if the message is sent successfully, double grey ticks if the message is received successfully by the recipient and double blue ticks if the message is seen or read by the recipient. This is shown below in an illustration by "Kuwaitiful":https://kuwaitiful.com/tag/whatsapp/.
+
+<a href="/images/miscellaneous/whatsapp-receipts.png" target="_blank">
+  <img src="/images/miscellaneous/whatsapp-receipts.png" style="width: 100%" alt="Read Receipts in WhatsApp">
+</a>
+
+This feature can be quite easily implemented using "Ably":https://ably.io. The implementation details are below.
+
+h2(#how). Implementation details
+
+Ably deals with messages in real time using named "channels":/realtime/channels-messages. For implementing read/delivery receipts in a chat app, in addition to the general messaging channels required for your use-case. These receipts need to be sent and received in real time, hence you need to use "Ably's Realtime Library":/realtime.
+
+For the read receipts, you can use a separate channel altogether, so that the messaging business logic is left undisturbed. Of course, we also need to know the "presence":/realtime/presence status of the users. So, for a generic chat app between two users, let us consider three channels as shown below. This is explained further.
```

Comment:
"so that the messaging business logic is left undisturbed." - Less business logic, and more just leaving any existing messaging structures undisturbed?